### ISSUE TITLE
Dynamically add lifecycle callbacks

### ISSUE THREAD

**Issue by jetsonjohn** [balderdashy/sails#3200]
I need to separate lifecycle callbacks from models and need to put it in another file and merge these callbacks to the model. 

So I created a hook and put these code in it. 

```javascript
module.exports = function lccb(sails) {
  return {
    initialize: function(cb) {
      sails.on('hook:orm:loaded', function() {
        sails.models.user['beforeValidate'] = function(values, cb) {
          console.log('beforeValidate');
          cb();
        };
      });
    }
  };
};
```

I printed the sails global object and I can see the beforecreate function in the model object but the function is not getting called. 

Is there any workaround to add the lifecycle callbacks from another file by using hooks or service

**Comment by jetsonjohn** [comment]
There is already fixed  #3165 but not in 0.11 stable version.

So I copied the core hook orm and place it in the under `api/hooks` and applied the changes mentioned in the pull request. Then it works.

Is it ok to override the core hook by copying the folder to `api/hooks`?

**Comment by sgress454** [comment]
> Is it ok to override the core hook by copying the folder to api/hooks?

Sounds like you know what you're getting yourself into, so yes, it is.  When v0.12 comes out, you'll probably want to remove the user hook and go back to using the core one, in case there are other updates!
